rioting-daisies/BangazonSite
# User can complete purchase

# Description

User can now complete an order by checking out and adding a payment option. Also implemented a soft delete for payment types so that data is not deleted from database but instead is hidden from user. Fixed the bug that allowed product quantity to be negative by adding a restraint on Quantity, and removing the add to cart button once quantity is 0.


Fixes Issue #50 

## Type of change

Please delete options that are not relevant.



# Testing Instructions

1. git fetch and checkout saturdaymorning. 
1. run update database to add the new migration.
1. Add products to your cart. Check to confirm that the product will be shown as sold out if the quantity goes to 0. 
1. navigate to the shopping cart, click checkout, add payment type, and complete purchase. the order should now show completed date, and shopping cart will be empty.  If there are no payment types, user will be redirected to PaymentType create page. 



# Checklist:
